src/common/menu.rs

```diff
@@ -272,3 +256,196 @@ pub fn handle_get_npad_state(state: *mut NpadGcState, _controller_id: *const u32
+                app.on_right();
+                received_input = true;
+            });
+            button_presses.up.read_press().then(|| {
+                app.on_up();
+                received_input = true;
+            });
+            button_presses.down.read_press().then(|| {
+                app.on_down();
+                received_input = true;
+            });
+
+            std::thread::sleep(std::time::Duration::from_millis(16));
+            has_slept_millis += 16;
+            if has_slept_millis < 16 * render_frames {
+                continue;
+            }
+            has_slept_millis = 16;
+            if !menu::QUICK_MENU_ACTIVE {
+                app = training_mod_tui::App::new(consts::get_menu());
+                set_should_display_text_to_screen(false);
+                continue;
+            }
+            if !received_input {
+                continue;
+            }
+            let mut view = String::new();
+
+            let frame_res = terminal
+                .draw(|f| json_response = training_mod_tui::ui(f, &mut app)) // Ensure this gets updated to JSON
+                .unwrap();
+
+            use std::fmt::Write;
+            for (i, cell) in frame_res.buffer.content().iter().enumerate() {
+                match cell.fg {
+                    Color::Black => write!(&mut view, "{}", &cell.symbol.black()),
+                    Color::Blue => write!(&mut view, "{}", &cell.symbol.blue()),
+                    Color::LightBlue => write!(&mut view, "{}", &cell.symbol.bright_blue()),
+                    Color::Cyan => write!(&mut view, "{}", &cell.symbol.cyan()),
+                    Color::LightCyan => write!(&mut view, "{}", &cell.symbol.cyan()),
+                    Color::Red => write!(&mut view, "{}", &cell.symbol.red()),
+                    Color::LightRed => write!(&mut view, "{}", &cell.symbol.bright_red()),
+                    Color::LightGreen => write!(&mut view, "{}", &cell.symbol.bright_green()),
+                    Color::Green => write!(&mut view, "{}", &cell.symbol.green()),
+                    Color::Yellow => write!(&mut view, "{}", &cell.symbol.yellow()),
+                    Color::LightYellow => write!(&mut view, "{}", &cell.symbol.bright_yellow()),
+                    Color::Magenta => write!(&mut view, "{}", &cell.symbol.magenta()),
+                    Color::LightMagenta => {
+                        write!(&mut view, "{}", &cell.symbol.bright_magenta())
+                    }
+                    _ => write!(&mut view, "{}", &cell.symbol),
+                }
+                .unwrap();
+                if i % frame_res.area.width as usize == frame_res.area.width as usize - 1 {
+                    writeln!(&mut view).unwrap();
+                }
+            }
+            writeln!(&mut view).unwrap();
+
+            render_text_to_screen(view.as_str());
+            received_input = false;
+        }
+    }
+}
+
+static mut SHOULD_SHOW_MENU: bool = false;
```

Comment:
Rename to `SHOULD_SHOW_WEB_MENU`? Or, to be in line with the naming of the quick menu which follows a similar design pattern overall, `WEB_MENU_ACTIVE`?